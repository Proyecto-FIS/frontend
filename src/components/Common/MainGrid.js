import { Grid } from "@material-ui/core";

const MainGrid = ({ children }) => (
    <Grid container>
        <Grid item sm={2} xs={1}></Grid>
        <Grid item sm={8} xs={10}>
            {children}
        </Grid>
        <Grid item sm={2} xs={1}></Grid>
    </Grid>
);

export default MainGrid;
