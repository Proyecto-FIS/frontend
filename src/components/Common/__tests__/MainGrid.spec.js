import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, screen } from "@testing-library/react";
import MainGrid from "../MainGrid";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("empty render", () => {
    render(<MainGrid />);
});

it("with internal content", () => {
    render(
        <MainGrid>
            <div>Hello!</div>
        </MainGrid>
    );

    const elem = screen.getByText("Hello!");
    expect(elem).toBeInTheDocument();
});

it("matches snapshot", () => {
    const tree = renderer
        .create(<MainGrid attr="value"><h1>Hello!</h1></MainGrid>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
