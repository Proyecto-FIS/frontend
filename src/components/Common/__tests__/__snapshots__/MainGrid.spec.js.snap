// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-2"
  />
  <div
    attr="value"
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-8"
  >
    <h1>
      Hello!
    </h1>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-sm-2"
  />
</div>
`;
